pkg/controller/jobs/job/job_controller_test.go

```diff
@@ -402,6 +416,8 @@ func TestReconciler(t *testing.T) {
 		wantJob           batchv1.Job
 		wantWorkloads     []kueue.Workload
 		wantErr           error
+		checkWlOwner      bool
```

Comment:
I would suggest always checking

Reply:
I see, but:
1. `checkWlOwner` is set true in tandem with `skipOwnerUpdates`, we could just have one param `setWorkloadOwnershipInTest` (or alike)
2. we can infer the the value of `setWorkloadOwnershipInTest` based on `PrebuiltWorkloadLabel` set on job.

I think eliminating 2 params (or at least 1) from the test options is a win. First, we maintain the same coverage in non-prebuilt workload scenario. Second, we test that the owner is set in the prebuilt-workload scenario.

Reply:
Hm, sorry, I didn't see in this test "when the prebuilt workload is missing, no new one is created and the job is suspended" it is not in tandem.

Still, something feels over-complicated for the test setup.

Reply:
Maybe it is a sign this should be a dedicated unit test. I also don't see anything wrong in asserting on the owners even if they are set in the test code `L1835-L1837`. Ok, it might be unnecessary, but allows to eliminate a param, so feels like a win. Also, we would verify also that, for example, the extra reference is not added.

Anyway, the current version is acceptable to me.